File: MATLAB/Laplacian Kernel.m
kernelSize = 3;  
laplacianKernel = -4 * ones(kernelSize) + eye(kernelSize) * (kernelSize^2 - 1);
inputImage = imread('Enter the path to the image here');
if size(inputImage, 3) == 3
    inputImage = rgb2gray(inputImage);
end
filteredImage = conv2(double(inputImage), laplacianKernel, 'same');
filteredImage = mat2gray(filteredImage);
figure('Position', [100, 100, 600, 800]);
subplot(2, 1, 1);
imshow(inputImage);
title('Original Image');
subplot(2, 1, 2);
imshow(filteredImage);
title('Laplacian Filtered Image');

File: Python/Laplacian Kernel.py
import cv2
import numpy as np
import matplotlib.pyplot as plt

# Kernel size
kernel_size = 3

# Create Laplacian kernel
laplacian_kernel = -4 * np.ones((kernel_size, kernel_size))
laplacian_kernel += np.eye(kernel_size) * (kernel_size**2 - 1)

# Read the image
input_image = cv2.imread('D:\\Academics\Semesters\\4-1\\Lab\\Digital Image Processing Laboratory\\Programs\\Lab4\\stray.jpg', cv2.IMREAD_GRAYSCALE)

# Apply convolution
filtered_image = cv2.filter2D(src=input_image.astype(np.float32), ddepth=-1, kernel=laplacian_kernel)

# Normalize the result to 0-1 range
filtered_image = cv2.normalize(filtered_image, None, alpha=0, beta=1, norm_type=cv2.NORM_MINMAX)

# Plot the results
plt.figure(figsize=(6, 8))
plt.subplot(2, 1, 1)
plt.imshow(input_image, cmap='gray')
plt.title('Original Image')
plt.axis('off')

plt.subplot(2, 1, 2)
plt.imshow(filtered_image, cmap='gray')
plt.title('Laplacian Filtered Image')
plt.axis('off')

plt.tight_layout()
plt.show()
